src/openpgp/packet/keymaterial.iced

```diff
@@ -461,15 +461,51 @@ class KeyMaterial extends Packet
 
   # Returns non-zero expire time if it exists, otherwise null.
   get_expire_time : () ->
-    return null unless (psc = @get_psc())?
-    list = psc.lookup[if @is_primary() then "self_sig" else "subkey_binding"]
+    if not (psc = @get_psc())? then null
+    else if @is_primary() then @_get_expire_time_on_primary()
+    else @_get_expire_time_on_subkey()
+
+  #-------------------
+
+  _get_expire_time_on_primary : () ->
+    table = @get_psc().lookup.self_sigs_by_uid
+
+    winner = null
+    key_generated = @timestamp
+
+    for uid,list of table
+
+      uid_winner = null
+      for packetsig in list when (sig = packetsig.sig)?
```

Comment:
Good Q.  I'm not sure, but I copied the condition from the previous code used for both primary and subkey.